File: Python/Concepts/C Sharp/InsertionSort.cs
public int[] insertionSort(int[] A){
    int n = A.Length;
    for(int i = 1; i < n;i++){
        int key = A[i];
        int j = i-1;
        while(j >= 0 && key < A[j]){
            A[j+1] = A[j];
            j--;
        }
        A[j+1] = key;
    }
    return A;
}

@"
TC = O(n);
SC = O(1);
Stable
"

File: Python/Concepts/InsertionSort.py
def insertion_sort(array):
    for i in range(1, len(array)):
        key = array[i]
        j = i - 1
        while j >= 0 and key < array[j]:
            array[j + 1] = array[j]
            j -= 1
        array[j + 1] = key
        print(array)
    return array


array = [8, 5, 2, 9, 6, 3]
insertion_sort(array)
